hexrdgui/main_window.py

```diff
@@ -1730,3 +1735,55 @@ def action_apply_median_filter_toggled(self, b):
         # The dialog should have modified HexrdConfig's median filter options
         # already. Just apply it now.
         HexrdConfig().apply_median_filter_correction = b
+
+    def validateDragDropEvent(self, event):
+        mime_data = event.mimeData()
+        if not mime_data.hasUrls():
+            event.ignore()
+            return
+
+        # Make sure the selction(s) are files with a supported file extensions
+        paths = [url.toLocalFile() for url in mime_data.urls()]
+        supported_extensions = [
+            '.hexrd',
+            '.yml', '.yaml',
+            '.h5', '.hdf5',
+            '.tif', '.tiff',
+            '.png',
+            '.jpg', '.jpeg'
+        ]
+        exts = [Path(path).suffix.lower() for path in paths]
+        if any(ext not in supported_extensions for ext in exts):
+            event.ignore()
+            return
+
+        if event.type() == QEvent.Drop:
+            self.dropEvent(paths)
+        else:
+            event.acceptProposedAction()
+
+    def dropEvent(self, paths):
+        ext = Path(paths[0]).suffix.lower()
+        if len(paths) == 1:
+            try:
+                # if hdf5 try as state, then materials, then images
+                if ext in ('.h5', '.hdf5'):
+                    try:
+                        # Try loading it as a state file first
+                        self.load_state_file(paths[0])
+                    except:
+                        # If that fails, try loading it as a materials file
+                        HexrdConfig().load_materials(paths[0])
+                elif ext in ('.hexrd', '.yml', '.yaml'):
+                    try:
+                        # Try loading it as an instrument config first
+                        HexrdConfig().load_instrument_config(paths[0])
+                    except:
+                        # If that fails, try loading it as an image file
+                        self.open_image_files(selected_files=paths)
```

Comment:
```suggestion
                    # Try loading it as an instrument config first
                    HexrdConfig().load_instrument_config(paths[0])
```

There's already a `try/except` block outside of this that tries to open as image files.